### ISSUE TITLE
[Bug]: HMR issues on renaming stories

### ISSUE THREAD

**Issue by vanessayuenn** [storybookjs/storybook#25788]
### Describe the bug

originally filed [here](https://github.com/storybookjs/storybook/discussions/25687#discussioncomment-8245240)

**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

When renaming the story file of which you are visiting, a lot of weird things happen, it's best described in a video:

https://github.com/storybookjs/storybook/assets/1671563/540e03b8-82bf-4893-a0e3-adfe8058cf3f

The issues which happen are:
1 - The story of which you renamed somehow changes order and goes to the bottom of the list (there is no storySort being used)
2 - An `importers[path] is not a function` error is flashed in the screen:
<img width="349" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/5c5dc9c3-40c1-4252-a6fb-da45a5f9768f">

3 - The UI goes into a funky state where there's some sort of negative margin and the toolbar is unreachable (until you close the addon panel, which is not really closed)

### To Reproduce

1. Go to https://github.com/yannbf/mealdrop/tree/storybook/8.0.0
2. Visit a story
3. Rename its story file to something else (e.g. add an A at the end)

**Comment by dziamid** [comment]
Yeah, currently refactoring stories (renaming or moving story files or exports in story files0 is a pain. I basically need to restart SB after every refactor.
